{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.7.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"pip install gdown","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","execution":{"iopub.status.busy":"2022-12-01T16:23:11.218576Z","iopub.execute_input":"2022-12-01T16:23:11.218963Z","iopub.status.idle":"2022-12-01T16:23:20.301560Z","shell.execute_reply.started":"2022-12-01T16:23:11.218927Z","shell.execute_reply":"2022-12-01T16:23:20.300467Z"},"trusted":true},"execution_count":2,"outputs":[{"name":"stdout","text":"Requirement already satisfied: gdown in /opt/conda/lib/python3.7/site-packages (4.5.4)\nRequirement already satisfied: six in /opt/conda/lib/python3.7/site-packages (from gdown) (1.15.0)\nRequirement already satisfied: tqdm in /opt/conda/lib/python3.7/site-packages (from gdown) (4.64.0)\nRequirement already satisfied: beautifulsoup4 in /opt/conda/lib/python3.7/site-packages (from gdown) (4.11.1)\nRequirement already satisfied: filelock in /opt/conda/lib/python3.7/site-packages (from gdown) (3.7.1)\nRequirement already satisfied: requests[socks] in /opt/conda/lib/python3.7/site-packages (from gdown) (2.28.1)\nRequirement already satisfied: soupsieve>1.2 in /opt/conda/lib/python3.7/site-packages (from beautifulsoup4->gdown) (2.3.1)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests[socks]->gdown) (2022.9.24)\nRequirement already satisfied: charset-normalizer<3,>=2 in /opt/conda/lib/python3.7/site-packages (from requests[socks]->gdown) (2.1.0)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests[socks]->gdown) (1.26.12)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests[socks]->gdown) (3.3)\nRequirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /opt/conda/lib/python3.7/site-packages (from requests[socks]->gdown) (1.7.1)\n\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0mNote: you may need to restart the kernel to use updated packages.\n","output_type":"stream"}]},{"cell_type":"code","source":"import gdown\nimport os\n\nurl = 'https://drive.google.com/file/d/1OQbkhgcOAKv1brPE50QhTw6TpwWsgMAo/view?usp=share_link'\n\noutput = 'profiles_brands.zip'\n# os.remove(\"/kaggle/working/profiles_brands.zip\")\ngdown.download(url, output, quiet=False)","metadata":{"execution":{"iopub.status.busy":"2022-12-01T16:23:32.998513Z","iopub.execute_input":"2022-12-01T16:23:32.998870Z","iopub.status.idle":"2022-12-01T16:23:33.688332Z","shell.execute_reply.started":"2022-12-01T16:23:32.998839Z","shell.execute_reply":"2022-12-01T16:23:33.687377Z"},"trusted":true},"execution_count":4,"outputs":[{"name":"stderr","text":"/opt/conda/lib/python3.7/site-packages/gdown/parse_url.py:39: UserWarning: You specified a Google Drive link that is not the correct link to download a file. You might want to try `--fuzzy` option or the following url: https://drive.google.com/uc?id=1OQbkhgcOAKv1brPE50QhTw6TpwWsgMAo\n  url=\"https://drive.google.com/uc?id={}\".format(file_id)\nDownloading...\nFrom: https://drive.google.com/file/d/1OQbkhgcOAKv1brPE50QhTw6TpwWsgMAo/view?usp=share_link\nTo: /kaggle/working/profiles_brands.zip\n71.5kB [00:00, 9.17MB/s]\n","output_type":"stream"},{"execution_count":4,"output_type":"execute_result","data":{"text/plain":"'profiles_brands.zip'"},"metadata":{}}]},{"cell_type":"code","source":"import sys\nimport json\nimport pandas as pd\nimport random\nimport numpy as np\n\n\nif sys.version_info >= (3, 6):\n    import zipfile\nelse:\n    import zipfile36 as zipfile\n\ncols = ['id', 'shortcode', 'owner.id', '__typename', 'is_video', 'comments_disabled', 'dimensions.height', 'dimensions.width', 'edge_media_preview_like.count', \n'edge_media_to_comment.count', 'caption_text', 'taken_at_timestamp']\n\ndfs = []\nfile_name = 'this_is_my_path/File_A.json'\n\n\n# counter = 0;\n#         print(counter) \n            \n#         if( counter == 7 ):\n#             break\n#         counter += 1\n\ndtypes_dict = {\n    'id': 'int64',\n    'shortcode': 'str',\n    'owner.id': 'int64',\n    '__typename': 'str',\n    'is_video': 'bool',\n    'comments_disabled': 'bool',\n    'dimensions.height': 'int32',\n    'dimensions.width': 'int32',\n    'edge_media_preview_like.count': 'int32', \n    'edge_media_to_comment.count': 'int32',\n    'caption_text': 'str',\n    'taken_at_timestamp': 'str',    \n }\n\nwith zipfile.ZipFile('/kaggle/working/profile_brands.zip', 'r') as z:\n    for i in z.infolist():\n\n        if not i.is_dir() and i.filename.endswith('.file'):    \n      \n            with z.open((i.filename), 'r') as f:\n                \n                data = f.read()\n                json_file = json.loads(data)\n\n                df = pd.json_normalize(json_file)\n                if(df['edge_media_to_caption.edges'][0]):\n                    df['caption_text'] = df['edge_media_to_caption.edges'][0][0]['node']['text']\n                else:\n                    df['caption_text'] = \"\"\n                # print(df['caption_text'])\n                \n                try:\n                    df = df[cols]\n                    df = df.astype(dtypes_dict)\n                    arr = df.to_numpy()\n                    dfs.append(arr[0]) \n                    \n                except KeyError as ex:\n                    continue       \n\n","metadata":{},"execution_count":null,"outputs":[]}]}